FrontEndReact/src/View/Login/SetNewPassword.js

```diff
@@ -3,87 +3,277 @@ import 'bootstrap/dist/css/bootstrap.css';
+                    break;
+                case 'WEAK':
+                    icon = ErrorOutlineIcon;
+                    break;
+                case 'MEDIUM':
+                    icon = ErrorOutlineIcon;
+                    break;
+                case 'STRONG':
+                    icon = CheckIcon;
+                    break;
+            }
+
+            return icon;
+        };
+
+        this.generateColors = (strength) => {
+            let result = [];
+
+            const COLORS = {
+                NEUTRAL: '#E2E2E2',
+                WEAK: '#B40314',
+                MEDIUM: '#D39323',
+                STRONG: '#7B927F',
+            };
+        
+            switch(strength) {
+                default: 
+                    result = [COLORS.WEAK, COLORS.NEUTRAL, COLORS.NEUTRAL, COLORS.NEUTRAL];
+                    break;
+                case 'WEAK':
+                    result = [COLORS.WEAK, COLORS.NEUTRAL, COLORS.NEUTRAL, COLORS.NEUTRAL];
+                    break;
+                case 'MEDIUM':
+                    result = [COLORS.MEDIUM, COLORS.MEDIUM, COLORS.NEUTRAL, COLORS.NEUTRAL];
+                    break;
+                case 'STRONG':
+                    result = [COLORS.STRONG, COLORS.STRONG, COLORS.STRONG, COLORS.STRONG];
+                    break;
+            }
+        
+            return result;
+        };
+
+        this.testingPasswordStrength = (password) => {
+            const atLeastMinimumLength = (password) => new RegExp(/(?=.{8,})/).test(password);
+            const atLeastOneUppercaseLetter = (password) => new RegExp(/(?=.*?[A-Z])/).test(password);
+            const atLeastOneLowercaseLetter = (password) => new RegExp(/(?=.*?[a-z])/).test(password);
+            const atLeastOneNumber = (password) => new RegExp(/(?=.*?[0-9])/).test(password);
+            const atLeastOneSpecialChar = (password) => new RegExp(/(?=.*?[#?!@$%^&*-])/).test(password);
+
+            if (!password) return 'WEAK';
+
+            let points = 0;
+
+            if (atLeastMinimumLength(password)) points += 1;
+            if (atLeastOneUppercaseLetter(password)) points += 1;
+            if (atLeastOneLowercaseLetter(password)) points += 1;
+            if (atLeastOneNumber(password)) points += 1;
+            if (atLeastOneSpecialChar(password)) points += 1;
+        
+            if (points >= 5) return 'STRONG';
+            if (points >= 3) return 'MEDIUM';
+
+            return 'WEAK';
         }
+        
+
         this.setPassword = () => {
-            var pass1 = document.getElementById("password").value;
-            var pass2 = document.getElementById("password2").value;
-    
-            if (pass1 === pass2) {
+            var pass1 = this.state.password;
+            var pass2 = this.state.password2;
 
+            var passwordSecurity = this.testingPasswordStrength(pass1);
+
+            if (pass1 === pass2 && passwordSecurity === "STRONG") {
                 var valid = validPasword(pass1);
```

Comment:
Is this line needed anymore?

Reply:
I committed changes to remove unnecessary variable valid!